core/src/main/scala/kafka/zk/ZkData.scala

```diff
@@ -281,7 +281,7 @@ object TopicZNode {
           new TopicPartition(topic, partition.toInt) -> PartitionReplicaAssignment(
             replicas.to[Seq[Int]],
             getReplicas(addingReplicasJsonOpt, partition),
-            getReplicas(addingReplicasJsonOpt, partition)
+            getReplicas(removingReplicasJsonOpt, partition)
```

Comment:
Do we have a test for `decode`? If not, we should.

Reply:
We didn't, I added one with 91572b79fe564e1810fe3e7f4696501098c9a5f4 - thanks

Reply:
Interesting. Is it fair to stay that all of the tests passed because the partition leader is ignoring `removingReplicas`? Same of the controller, the controller only waits for `addingReplicas` to join the ISR?

Reply:
The partition leaders do not currently make use of the new removing/adding replicas fields. When receiving a LAIR, they simply start treating all of `replicas` as replicas

We did not catch this because we were lacking an assertion in an integration test that reloads the reassignment state from ZK - https://github.com/apache/kafka/pull/7334/files#diff-386e7e8ca0062980d6c88c2ae9c055f0R687